test/it/controller/etcdopstask/helper.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcdopstask
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	druidconfigv1alpha1 "github.com/gardener/etcd-druid/api/config/v1alpha1"
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask/handler"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask/handler/ondemandsnapshot"
+	"github.com/gardener/etcd-druid/test/it/setup"
+	testutils "github.com/gardener/etcd-druid/test/utils"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	. "github.com/onsi/gomega"
+)
+
+// ReconcilerTestEnv encapsulates the test environment required to test the EtcdOpsTask reconciler.
+type ReconcilerTestEnv struct {
+	itTestEnv  setup.DruidTestEnvironment
+	reconciler *etcdopstask.Reconciler
+}
+
+// InjectHTTPClient injects an HTTP client into the reconciler's task handler registry for testing.
+func (env *ReconcilerTestEnv) InjectHTTPClient(httpClient *http.Client) {
```

Comment:
This is no longer required when you move to using `NewReconcilerWithTaskHandlerRegistry` constructor function.